compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1551,7 +1551,7 @@ impl<'a> Parser<'a> {
                 })
             });
             consume_colon = false;
-            Ok(self.mk_expr(lo, ExprKind::Lit(lit.token_lit)))
+            Ok(self.mk_expr(lo, ExprKind::Lit(lit.as_token_lit())))
```

Comment:
Also suspicious, if we are in the parser we should not use `MetaItemLit`.

Reply:
Fixed.